input_explainer.md

```diff
@@ -0,0 +1,406 @@
+
+As 3D mouse cursors require such an app-driven feedback loop with scene content, tracking such a cursor based on mouse motion is out of scope for `pointerPoseMatrix` as currently defined.  Instead, mouse left-clicks cause `select` gestures with a gaze-targeted or Magic Window `pointerPoseMatrix`.  This matches all other contextless input sources described above, where the UA tosses a single natural pointing ray over the wall to the app without knowledge of the scene content, similar to traditional 2D mouse UI.  
+
+Apps or middleware may then choose to implement a 3D mouse cursor by requesting pointer lock and then handling `mouseMove` events to accumulate a custom 3D cursor position over time, reacting to scene content as they see fit.
+
+TODO: Rationalize with onvrdisplaypointerrestricted from WebVR 1.1.
+
+## Appendix A: I don’t understand why this is a new API. Why can’t we use…
+
+### Pointer events
+Pointer events and their various sub-parts (mouse, touch, etc.) were created with the needs of a 2D web in mind, and serve that purpose well. Trying to bolt on an understanding of 3D space would both over-complicate the APIs with functionality that most applications don't need and make VR/AR input a second-class citizen within the larger pointer even model.
+
+We do expect that many future VR/AR uses (such as interacting with 2D DOM rectangles in 3D space) will need to emulate 2D input methods, producing appropriately transformed pointer events in response to VR/AR input.
+
+### The Gamepad API
+This was actually the first thing that we tried, extending the API with the necessary pose data. It proved to be mechanically feasible, but provided a confusing, non-semantic view of VR/AR input that left developers guessing as to what each button meant based on the device name string. Obviously that's a fairly fragile model, and not one that can reasonably extend to a robust ecosystem of hundreds of different devices. It also lacks many of the higher level concepts that we feel developers will eventually need (events for user initiation, ray generation, action mapping across devices, controller visualization, etc.)
+
+## Appendix B: Proposed IDL
+
+```webidl
+//
+// Controllers
+//
+
+interface VRInputSource {
+};
+
+interface VRGamepadInputSource : VRInputSource {
+  readonly attribute Gamepad gamepad;
+};
+
+enum VRHandedness {
+  "",
+  "left",
+  "right"
+};
+
+interface VRController : VRInputSource {
+  readonly attribute VRHandedness handedness;
+
+  // TODO: Better names for these two properties.
+  readonly attribute boolean supportsPointing; 
```

Comment:
This comes from my perspective of higher level APIs so not sure if this is relevant in this context. If we expect to consider more gestures in the future it might be cleaner to return an object to be able to do:
````javascript
controller.gestures.pointing
````